### PR TITLE
Test more versions of Jenkins core

### PR DESCRIPTION
**Pull request by MarkEWaite**
## Test more versions of Jenkins core

- Test with Jenkins 2.361.3 (Java 11) and 2.375 (Java 17)
- Fix Jenkins 2.375 compile failure

### CODE REVIEW

File: src/main/java/com/dabsquared/gitlabjenkins/GitLabPushTrigger.java
Comment (jglick) [inline src/main/java/com/dabsquared/gitlabjenkins/GitLabPushTrigger.java]: (subsuming #1358)

Comment (MarkEWaite) [reply src/main/java/com/dabsquared/gitlabjenkins/GitLabPushTrigger.java]: I think that I will merge #1358 once it completes its CI run so that the changelog includes the text from #1358.  I merged it into this pull request to assure that this pull request would also verify.

Comment (jglick) [reply src/main/java/com/dabsquared/gitlabjenkins/GitLabPushTrigger.java]: FYI, topologically (not squash- or rebase-) merging this PR would cause #1358 to be considered merged as well, I think according to Release Drafter too.

Comment (jglick) [reply src/main/java/com/dabsquared/gitlabjenkins/GitLabPushTrigger.java]: Basically, a PR is “merged” when _either_
* it was squash- or rebase-merged by GitHub; or
* its head commit becomes an ancestor of its base branch, by any means (including e.g. `git merge` without any GH-specific tools).

**Comment by MarkEWaite** [comment]
@gounthar or @jmMeessen could you approve this pull request?  Sorry to bother you on a holiday, but it is easier if another maintainer approves the pull request.  If no approval is received, I'll override and merge it myself without the extra approval.

**Review by basil — APPROVED** [review]

**Pull request merged**
